### PR TITLE
24.04/add some libboost 1.83 slices

### PR DESCRIPTION
**Pull request by endersonmaia**
# Proposed changes

As part of the issue #244 , this PR will add some of the packages.

## Related issues/PRs

#224

## Testing

```bash
chisel cut \
    --release ./ \
    --root rootfs \
    libboost-coroutine1.83.0_libs libboost-context1.83.0_libs libboost-filesystem1.83.0_libs libboost-log1.83.0_libs
2024/05/10 11:58:50 Processing ./ release...
2024/05/10 11:58:50 Selecting slices...
2024/05/10 11:58:50 Fetching ubuntu 24.04 noble suite details...
2024/05/10 11:58:50 Release date: Thu, 25 Apr 2024 15:10:33 UTC
2024/05/10 11:58:50 Fetching index for ubuntu 24.04 noble main component...
2024/05/10 11:58:50 Fetching index for ubuntu 24.04 noble universe component...
2024/05/10 11:58:51 Fetching ubuntu 24.04 noble-security suite details...
2024/05/10 11:58:51 Release date: Thu, 09 May 2024 16:53:41 UTC
2024/05/10 11:58:51 Fetching index for ubuntu 24.04 noble-security main component...
2024/05/10 11:58:51 Fetching index for ubuntu 24.04 noble-security universe component...
2024/05/10 11:58:51 Fetching ubuntu 24.04 noble-updates suite details...
2024/05/10 11:58:51 Release date: Fri, 10 May 2024 10:20:47 UTC
2024/05/10 11:58:51 Fetching index for ubuntu 24.04 noble-updates main component...
2024/05/10 11:58:51 Fetching index for ubuntu 24.04 noble-updates universe component...
2024/05/10 11:58:51 Fetching pool/main/g/glibc/libc6_2.39-0ubuntu8.1_amd64.deb...
2024/05/10 11:58:51 Fetching pool/main/g/gcc-14/libgcc-s1_14-20240412-0ubuntu1_amd64.deb...
2024/05/10 11:58:51 Fetching pool/main/g/gcc-14/libstdc++6_14-20240412-0ubuntu1_amd64.deb...
2024/05/10 11:58:51 Fetching pool/main/b/boost1.83/libboost-context1.83.0_1.83.0-2.1ubuntu3_amd64.deb...
2024/05/10 11:58:51 Fetching pool/universe/b/boost1.83/libboost-coroutine1.83.0_1.83.0-2.1ubuntu3_amd64.deb...
2024/05/10 11:58:51 Fetching pool/main/b/boost1.83/libboost-filesystem1.83.0_1.83.0-2.1ubuntu3_amd64.deb...
2024/05/10 11:58:51 Fetching pool/main/b/boost1.83/libboost-thread1.83.0_1.83.0-2.1ubuntu3_amd64.deb...
2024/05/10 11:58:51 Fetching pool/universe/b/boost1.83/libboost-log1.83.0_1.83.0-2.1ubuntu3_amd64.deb...
2024/05/10 11:58:51 Extracting files from package "libc6"...
2024/05/10 11:58:51 Extracting files from package "libgcc-s1"...
❯ ldd rootfs/usr/lib/x86_64-linux-gnu/libboost_log.so.1.83.0
        linux-vdso.so.1 (0x00007ffeaf53d000)
        libboost_filesystem.so.1.83.0 => /lib/x86_64-linux-gnu/libboost_filesystem.so.1.83.0 (0x00007b1df1fc7000)
        libboost_thread.so.1.83.0 => /lib/x86_64-linux-gnu/libboost_thread.so.1.83.0 (0x00007b1df1fa6000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007b1df1c00000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007b1df1f79000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007b1df1800000)
        /lib64/ld-linux-x86-64.so.2 (0x00007b1df20b2000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007b1df1e8e000)
❯ ldd rootfs/usr/lib/x86_64-linux-gnu/libboost_log_setup.so.1.83.0
        linux-vdso.so.1 (0x00007fff85902000)
        libboost_log.so.1.83.0 => not found
        libboost_thread.so.1.83.0 => /lib/x86_64-linux-gnu/libboost_thread.so.1.83.0 (0x00007c6b0b33a000)
        libboost_filesystem.so.1.83.0 => /lib/x86_64-linux-gnu/libboost_filesystem.so.1.83.0 (0x00007c6b0b317000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007c6b0b000000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007c6b0b2ea000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007c6b0ac00000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007c6b0af17000)
        /lib64/ld-linux-x86-64.so.2 (0x00007c6b0b444000)
❯ ldd rootfs/usr/lib/x86_64-linux-gnu/libboost_coroutine.so.1.83.0
        linux-vdso.so.1 (0x00007fff6756f000)
        libboost_context.so.1.83.0 => not found
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007b61c2a00000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007b61c2600000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007b61c2cd2000)
        /lib64/ld-linux-x86-64.so.2 (0x00007b61c2dd8000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007b61c2ca5000)

```

## Checklist

* [x] I have read the [contributing guidelines](
https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md)
* [x] I have already submitted the [CLA form](
https://ubuntu.com/legal/contributors/agreement)

### CODE REVIEW

File: slices/libboost-coroutine1.83.0.yaml
Comment (zhijie-yang) [inline slices/libboost-coroutine1.83.0.yaml:1]: This depends on libgcc-s1 for armhf.

**Review by zhijie-yang — CHANGES_REQUESTED** [review]
Looks good to me. Just to notice coroutine requires libgcc-s1 for armhf machines.

File: slices/libboost-coroutine1.83.0.yaml
Comment (rebornplusplus) [reply slices/libboost-coroutine1.83.0.yaml:1]: We will need https://github.com/canonical/chisel/issues/93 for this.

I guess the best way forward would be to add a comment in the slice definition file, instructing to install `libgcc-s1_libs` separately for armhf.

**Review by rebornplusplus — CHANGES_REQUESTED** [review]
Hi, thanks for the PR! The slices mostly look good. I reckon there are a few linting issues in this PR. It should be good to merge after those are fixed. Check the CI.

**Comment by endersonmaia** [comment]
I sent fixups, but the CI is failing, asking to sort on files I didn't changed.

Maybe the CI should only test the files that are in the PR?

File: slices/libboost-coroutine1.83.0.yaml
Comment (endersonmaia) [reply slices/libboost-coroutine1.83.0.yaml:1]: Sent a fixup with the comment at a3868c33e9a6b8456200ae3ab68d745e561d10bd

**Review by rebornplusplus — APPROVED** [review]
Looks good to me, thank you!

**Comment by endersonmaia** [comment]
> @endersonmaia could you pls double check your email in LP? The CLA check is failing

Weird, it was passing before, right?

I'll try to see what's going on.

**Comment by endersonmaia** [comment]
> @endersonmaia could you pls double check your email in LP? The CLA check is failing

Everything looks ok with my LP account.

**Review by cjdcordeiro — APPROVED** [review]
This was indeed passing before and I can see your LP account has the right email. I can also see you are in the CLA list. This might be a glitch, so let's proceed anyway

**Pull request merged**